src/main/java/com/conveyal/datatools/manager/controllers/api/GtfsPlusController.java

```diff
@@ -230,21 +208,25 @@ private static Boolean publishGtfsPlusFile(Request req, Response res) {
         }
 
         FeedVersion newFeedVersion = new FeedVersion(feedVersion.parentFeedSource());
-
+        File newGtfsFile = null;
         try {
-            newFeedVersion.newGtfsFile(new FileInputStream(newFeed));
+            newGtfsFile = newFeedVersion.newGtfsFile(new FileInputStream(newFeed));
         } catch (IOException e) {
-            logMessageAndHalt(req, 500, "Error creating new FeedVersion from combined GTFS/GTFS+", e);
+            e.printStackTrace();
+            logMessageAndHalt(req, 500, "Error reading GTFS file input stream", e);
         }
+        if (newGtfsFile == null) {
+            logMessageAndHalt(req, 500, "GTFS input file must not be null");
+            return null;
+        }
+        newFeedVersion.fileSize = newGtfsFile.length();
+        newFeedVersion.hash = HashUtils.hashFile(newGtfsFile);
 
-        newFeedVersion.hash();
-
-        // validation for the main GTFS content hasn't changed
-        newFeedVersion.validationResult = feedVersion.validationResult;
-        newFeedVersion.storeUser(profile);
-        Persistence.feedVersions.create(newFeedVersion);
+        // Must be handled by executor because it takes a long time.
+        ProcessSingleFeedJob processSingleFeedJob = new ProcessSingleFeedJob(newFeedVersion, profile.getUser_id(), true);
+        DataManager.heavyExecutor.execute(processSingleFeedJob);
```

Comment:
The whole workflow of this method is unclear to me.  Why is the job being added when it wasn't there before?  This whole class could use a few more comments and this PR could use an issue explaining what is being fixed here.